code-challenges.test.js

```diff
@@ -14,60 +14,120 @@
+
+    const studyMinutesWeek2 = {
+      sunday: 100,
+      monday: 10,
+      tuesday: 45,
+      wednesday: 60,
+      thursday: 20,
+      friday: 15,
+      saturday: 65
+    }
+    // Expected output: [10, 15, 20, 45, 60, 65, 100]
+    expect(sortArray(studyMinutesWeek2)).toEqual([10, 15, 20, 45, 60, 65, 100])
+  })
+})
 
 // b) Create the function that makes the test pass.
 
-// Pseudo code:
+// Pseudo code
+// name : sort array
+// input : object
+// output : array
+// process : create a function that takes an object as a parameter, create a variable to store the objects values using the "object.values" method, then use .sort in order to create an array that displays the objects values in numarical order
+
+const sortArray = (object) => { // this function takes an object as a parameter
+  let studyMin = Object.values(object) // this stores the objects values in a new variable
+  studyMin.sort((a, b) => a - b) // this creates a new array with the objects values in numerical order
+  return studyMin
+}
 
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const accountTransactions1 = [100, -17, -23, -9]
-// Expected output: [100, 83, 60, 51]
+describe("userAccount", () => { // a test to check the funtion below
+  it("returns an array of the accumilating sum", () => {
 
-const accountTransactions2 = [250, -89, 100, -96]
-// Expected output: [250, 161, 261, 165]
+    const accountTransactions1 = [100, -17, -23, -9]
+    // Expected output: [100, 83, 60, 51]
+    expect(userAccount(accountTransactions1)).toEqual([100, 83, 60, 51])
 
-const accountTransactions3 = []
-// Expected output: []
+    const accountTransactions2 = [250, -89, 100, -96]
+    // Expected output: [250, 161, 261, 165]
+    expect(userAccount(accountTransactions2)).toEqual([250, 161, 261, 165])
+
+    const accountTransactions3 = []
+    // Expected output: []
+    expect(userAccount(accountTransactions3)).toEqual([])
+  })
+})
 
 // b) Create the function that makes the test pass.
 
-// Pseudo code:
+// Pseudo code
+// name : userAccount
+// input : 1 array
+// output : 1 array
+// process : create a function that takes an array as a parameter, create a conditional that checks if the array is empty, use .map and and have it add each of the subsequenting numbers of the array together.
+
+const userAccount = (array) => { // this function will take an array as a parameter
+  
+  let nextNum = 0 // this variable will be used in the map method and will represent the subsequenting number
+
+  if( array.length === 0) { // this will check if the array is emtpy 
+    return []
+  }
+  return array.map((value) => (nextNum += value)) // this will use .map to iterate through the array and add the subsequenting number by the last subsequenting number
```

Comment:
Wording is a little off here, would be better to say that you are adding the accumulating sum number to the current value, and reassigning the variable to the new sum of that arithmetic.